pym/portage/emaint/modules/sync/sync.py

```diff
@@ -122,9 +122,10 @@ def repo(self, **kwargs):
 			repos = repos.split()
 		available = self._get_repos(auto_sync_only=False)
 		selected = self._match_repos(repos, available)
-		if not selected:
-			msgs = [red(" * ") + "The specified repos were not found: %s" %
-				(bold(", ".join(repos))) + "\n   ...returning"]
+		if len(selected) < len(repos):
+			missing_repo_names = set(repos) - set(repo.name for repo in selected)
+			msgs = [red(" * ") + "The specified repo(s) were not found or have sync-type empty: %s" %
```

Comment:
I've rewritten the patch to make the checks in _get_repos(). Adding extra validation steps should be a lot easier now.